### ISSUE TITLE
ARM64 is it Possible

### ISSUE THREAD

**Issue by macooper59** [clemenko/rke_install_blog#11]
Hey there Clemenko,

Thank you for the Video and the article, but I see it is only for the AMD side of things. I am testng it on the Intel/AMD side but I use Raspberry Pi 4/5 & Orange Pi 5s also. I found a link to the [curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/arm64/kubectl -o /usr/local/bin/kubectl](url) and I am researching and testing for it. I need some help please. I was wondering if you had any ideas?

Can you point me in the right direction please.
Thank you,
Michael

**Comment by clemenko** [comment]
Hey Michael, Arm is possible. It all depends on what components you want though. Currently Rancher is experimental at best : https://ranchermanager.docs.rancher.com/v2.5/getting-started/installation-and-upgrade/advanced-options/enable-experimental-features/rancher-on-arm64. 
Hauler, kubectl and rke2 do support arm. The script can certainly be modified to accommodate for arm. 
Is this just for testing/playing?

**Comment by joshyorko** [comment]
@clemenko @macooper59

I can confirm the challenges with Rancher on ARM64 as being consistent with its experimental status. I’ve been testing Rancher in a mixed-node setup with Raspberry Pi 5 devices (ARM) and a Dell Precision laptop (x86_64), all running Ubuntu 24.04 LTS.

When ARM nodes are included, Rancher faces several issues, such as the Multi-Cluster Management Plugins failing to load and the inability to add cloud provider credentials (only S3 is available). Additionally, while Harvester successfully connects to Rancher, it enters an endless creation-deletion loop when attempting to provision VMs through Fleet, which I believe may be related to a TLS handshake error.

The same issues occur in a single-node setup and a fully ARM-based cluster consisting of two Raspberry Pi 5s and one Raspberry Pi 4, all running Ubuntu 24.04 LTS.

**Comment by macooper59** [comment]
I figured as much, I have been having issues with this for at least a year.

Thanks for the info,
Michael Cooper
*Certified Linux Professional*
*Docker/Kubernetes Trained*
RPi Hobby Blog <http://www.rpihobby.us>

**Comment by clemenko** [comment]
For this guide I had to stick with stable components. While RPi and other arm platforms are great, they are not widely adopted. Making it a hard target to QA. I do know that Rancher (Suse) is making great strides towards Arm all the things. But it will take some time. Since X86-64 SBCs are getting cheap and plentiful, are you able to test on those? Or VMs?

**Comment by macooper59** [comment]
Yes actually I can test them on the Raspberry Pi 5 and the Orange Pi 5. I
will have a beelink mini system with the N100 Processor soon and I am sure
it will work on that as well. I will do some testing and send you my
results. I can also test in a VM both on Intel/AMD and ARM. Just let me
know what you would like me to do.

If that's okay with you,
Michael Cooper
*Certified Linux Professional*
*Docker/Kubernetes Trained*
RPi Hobby Blog <http://www.rpihobby.us>

**Comment by clemenko** [comment]
Absolutely, would love to get feedback!

**Comment by clemenko** [comment]
Any update on testing with arm?

**Comment by clemenko** [comment]
Closing as per age.
